<!-- http%3A//en.wikipedia.org/wiki%3Faction%3Drender%26curid%3D477989%26oldid%3D601636932 2014-06-10-07-34-51 -->
<p>In <a href="//en.wikipedia.org/wiki/Bioinformatics" title="Bioinformatics">bioinformatics</a>, <b>sequence assembly</b> refers to <a href="//en.wikipedia.org/wiki/Sequence_alignment" title="Sequence alignment">aligning</a> and merging fragments of a much longer <a href="//en.wikipedia.org/wiki/DNA" title="DNA">DNA</a> sequence in order to reconstruct the original sequence. This is needed as <a href="//en.wikipedia.org/wiki/DNA_sequencing" title="DNA sequencing">DNA sequencing</a> technology cannot read whole genomes in one go, but rather reads small pieces of between 20 and 30000 bases, depending on the technology used. Typically the short fragments, called reads, result from <a href="//en.wikipedia.org/wiki/Shotgun_sequencing" title="Shotgun sequencing">shotgun sequencing</a> <a href="//en.wikipedia.org/wiki/Genome" title="Genome">genomic</a> DNA, or <a href="//en.wikipedia.org/wiki/Transcription_(genetics)" title="Transcription (genetics)">gene transcript</a> (<a href="//en.wikipedia.org/wiki/Expressed_sequence_tag" title="Expressed sequence tag">ESTs</a>).</p>
<p>The problem of sequence assembly can be compared to taking many copies of a book, passing each of them through a shredder with a different cutter, and piecing the text of the book back together just by looking at the shredded pieces. Besides the obvious difficulty of this task, there are some extra practical issues: the original may have many repeated paragraphs, and some shreds may be modified during shredding to have typos. Excerpts from another book may also be added in, and some shreds may be completely unrecognizable.</p>
<p></p>
<div id="toc" class="toc">
<div id="toctitle">
<h2>Contents</h2>
</div>
<ul>
<li class="toclevel-1 tocsection-1"><a href="#Genome_assemblers"><span class="tocnumber">1</span> <span class="toctext">Genome assemblers</span></a></li>
<li class="toclevel-1 tocsection-2"><a href="#EST_assemblers"><span class="tocnumber">2</span> <span class="toctext">EST assemblers</span></a></li>
<li class="toclevel-1 tocsection-3"><a href="#De-novo_vs._mapping_assembly"><span class="tocnumber">3</span> <span class="toctext">De-novo vs. mapping assembly</span></a></li>
<li class="toclevel-1 tocsection-4"><a href="#Influence_of_technological_changes"><span class="tocnumber">4</span> <span class="toctext">Influence of technological changes</span></a></li>
<li class="toclevel-1 tocsection-5"><a href="#Greedy_algorithm"><span class="tocnumber">5</span> <span class="toctext">Greedy algorithm</span></a></li>
<li class="toclevel-1 tocsection-6"><a href="#Available_assemblers"><span class="tocnumber">6</span> <span class="toctext">Available assemblers</span></a></li>
<li class="toclevel-1 tocsection-7"><a href="#See_also"><span class="tocnumber">7</span> <span class="toctext">See also</span></a></li>
<li class="toclevel-1 tocsection-8"><a href="#References"><span class="tocnumber">8</span> <span class="toctext">References</span></a></li>
</ul>
</div>
<p></p>
<h2><span class="mw-headline" id="Genome_assemblers">Genome assemblers</span></h2>
<p>The first sequence assemblers began to appear in the late 1980s and early 1990s as variants of simpler <a href="//en.wikipedia.org/wiki/Sequence_alignment" title="Sequence alignment">sequence alignment</a> programs to piece together vast quantities of fragments generated by automated sequencing instruments called <a href="//en.wikipedia.org/wiki/DNA_sequencers" title="DNA sequencers" class="mw-redirect">DNA sequencers</a>. As the sequenced organisms grew in size and complexity (from small <a href="//en.wikipedia.org/wiki/Viruses" title="Viruses" class="mw-redirect">viruses</a> over <a href="//en.wikipedia.org/wiki/Plasmids" title="Plasmids" class="mw-redirect">plasmids</a> to <a href="//en.wikipedia.org/wiki/Bacteria" title="Bacteria">bacteria</a> and finally <a href="//en.wikipedia.org/wiki/Eukaryotes" title="Eukaryotes" class="mw-redirect">eukaryotes</a>), the assembly programs used in these <a href="//en.wikipedia.org/wiki/Genome_project" title="Genome project">genome projects</a> needed increasingly sophisticated strategies to handle:</p>
<ul>
<li><a href="//en.wikipedia.org/wiki/Terabytes" title="Terabytes" class="mw-redirect">terabytes</a> of sequencing data which need processing on <a href="//en.wikipedia.org/wiki/Cluster_computing" title="Cluster computing" class="mw-redirect">computing clusters</a>;</li>
<li>identical and nearly identical sequences (known as <i>repeats</i>) which can, in the worst case, increase the time and space complexity of algorithms exponentially; and</li>
<li>errors in the fragments from the sequencing instruments, which can confound assembly.</li>
</ul>
<p>Faced with the challenge of assembling the first larger eukaryotic genomes—the fruit fly <a href="//en.wikipedia.org/wiki/Drosophila_melanogaster" title="Drosophila melanogaster">Drosophila melanogaster</a> in 2000 and the human genome just a year later,—scientists developed assemblers like Celera Assembler<sup id="cite_ref-1" class="reference"><a href="#cite_note-1"><span>[</span>1<span>]</span></a></sup> and Arachne<sup id="cite_ref-2" class="reference"><a href="#cite_note-2"><span>[</span>2<span>]</span></a></sup> able to handle genomes of 100-300 million base pairs. Subsequent to these efforts, several other groups, mostly at the major genome sequencing centers, built large-scale assemblers, and an open source effort known as AMOS<sup id="cite_ref-3" class="reference"><a href="#cite_note-3"><span>[</span>3<span>]</span></a></sup> was launched to bring together all the innovations in genome assembly technology under the <a href="//en.wikipedia.org/wiki/Open_source" title="Open source">open source</a> framework.</p>
<h2><span class="mw-headline" id="EST_assemblers">EST assemblers</span></h2>
<p><a href="//en.wikipedia.org/wiki/Expressed_Sequence_Tag" title="Expressed Sequence Tag" class="mw-redirect">Expressed Sequence Tag</a> or EST assembly differs from genome assembly in several ways. The sequences for EST assembly are the transcribed mRNA of a cell and represent only a subset of the whole genome. At a first glance, underlying algorithmical problems differ between genome and EST assembly. For instance, genomes often have large amounts of repetitive sequences, mainly in the inter-genic parts. Since ESTs represent gene transcripts, they will not contain these repeats. On the other hand, cells tend to have a certain number of genes that are constantly expressed in very high numbers (<a href="//en.wikipedia.org/wiki/Housekeeping_genes" title="Housekeeping genes" class="mw-redirect">housekeeping genes</a>), which again leads to the problem of similar sequences present in high numbers in the data set to be assembled.</p>
<p>Furthermore, genes sometimes overlap in the genome (<a href="//en.wikipedia.org/w/index.php?title=Sense-antisense_transcription&amp;action=edit&amp;redlink=1" class="new" title="Sense-antisense transcription (page does not exist)">sense-antisense transcription</a>), and should ideally still be assembled separately. EST assembly is also complicated by features like (cis-) <a href="//en.wikipedia.org/wiki/Alternative_splicing" title="Alternative splicing">alternative splicing</a>, <a href="//en.wikipedia.org/wiki/Trans-splicing" title="Trans-splicing">trans-splicing</a>, <a href="//en.wikipedia.org/wiki/Single-nucleotide_polymorphism" title="Single-nucleotide polymorphism">single-nucleotide polymorphism</a>, <a href="//en.wikipedia.org/wiki/Recoding" title="Recoding" class="mw-redirect">recoding</a>, and <a href="//en.wikipedia.org/wiki/Post-transcriptional_modification" title="Post-transcriptional modification">post-transcriptional modification</a>.</p>
<h2><span class="mw-headline" id="De-novo_vs._mapping_assembly">De-novo vs. mapping assembly</span></h2>
<p>In sequence assembly, two different types can be distinguished:</p>
<ol>
<li>de-novo: assembling short reads to create full-length (sometimes novel) sequences (see <a href="//en.wikipedia.org/wiki/De_novo_transcriptome_assembly" title="De novo transcriptome assembly">de novo transcriptome assembly</a>)</li>
<li>mapping: assembling reads against an existing backbone sequence, building a sequence that is similar but not necessarily identical to the backbone sequence</li>
</ol>
<p>In terms of complexity and time requirements, de-novo assemblies are orders of magnitude slower and more memory intensive than mapping assemblies. This is mostly due to the fact that the assembly algorithm needs to compare every read with every other read (an operation that has a naive time complexity of O(<var>n</var><sup>2</sup>); using a hash this can be reduced significantly). Referring to the comparison drawn to shredded books in the introduction: while for mapping assemblies one would have a very similar book as template (perhaps with the names of the main characters and a few locations changed), the de-novo assemblies are more hardcore in a sense as one would not know beforehand whether this would become a science book, a novel, a catalogue, or even several books. Also, every shred would be compared with every other shred.</p>
<h2><span class="mw-headline" id="Influence_of_technological_changes">Influence of technological changes</span></h2>
<p>The complexity of sequence assembly is driven by two major factors: the number of fragments and their lengths. While more and longer fragments allow better identification of sequence overlaps, they also pose problems as the underlying algorithms show quadratic or even exponential complexity behaviour to both number of fragments and their length. And while shorter sequences are faster to align, they also complicate the layout phase of an assembly as shorter reads are more difficult to use with repeats or near identical repeats.</p>
<p>In the earliest days of DNA sequencing, scientists could only gain a few sequences of short length (some dozen bases) after weeks of work in laboratories. Hence, these sequences could be aligned in a few minutes by hand.</p>
<p>In 1975, the <a href="//en.wikipedia.org/wiki/Dideoxy_termination" title="Dideoxy termination" class="mw-redirect">Dideoxy termination</a> method (also known as <a href="//en.wikipedia.org/wiki/Microfluidic_Sanger_Sequencing" title="Microfluidic Sanger Sequencing" class="mw-redirect"><i>Sanger sequencing</i></a>) was invented and until shortly after 2000, the technology was improved up to a point where fully automated machines could churn out sequences in a highly parallelised mode 24 hours a day. Large genome centers around the world housed complete farms of these sequencing machines, which in turn led to the necessity of assemblers to be optimised for sequences from whole-genome <a href="//en.wikipedia.org/wiki/Shotgun_sequencing" title="Shotgun sequencing">shotgun sequencing</a> projects where the reads</p>
<ul>
<li>are about 800–900 bases long</li>
<li>contain sequencing artifacts like sequencing and <a href="//en.wikipedia.org/wiki/Cloning_vectors" title="Cloning vectors" class="mw-redirect">cloning vectors</a></li>
<li>have error rates between 0.5 and 10%</li>
</ul>
<p>With the Sanger technology, bacterial projects with 20,000 to 200,000 reads could easily be assembled on one computer. Larger projects, like the human genome with approximately 35 million reads, needed large computing farms and distributed computing.</p>
<p>By 2004 / 2005, <a href="//en.wikipedia.org/wiki/Pyrosequencing" title="Pyrosequencing">pyrosequencing</a> had been brought to commercial viability by <a href="//en.wikipedia.org/wiki/454_Life_Sciences" title="454 Life Sciences">454 Life Sciences</a>. This new sequencing method generated reads much shorter than those of Sanger sequencing: initially about 100 bases, now 400-500 bases. Its much higher throughput and lower cost (compared to Sanger sequencing) pushed the adoption of this technology by genome centers, which in turn pushed development of sequence assemblers that could efficiently handle the read sets. The sheer amount of data coupled with technology-specific error patterns in the reads delayed development of assemblers; at the beginning in 2004 only the <a href="//en.wikipedia.org/wiki/Newbler" title="Newbler">Newbler</a> assembler from 454 was available. Released in mid-2007,<sup id="cite_ref-4" class="reference"><a href="#cite_note-4"><span>[</span>4<span>]</span></a></sup> the hybrid version of the MIRA assembler by Chevreux et al. was the first freely available assembler that could assemble 454 reads as well as mixtures of 454 reads and Sanger reads. Assembling sequences from different sequencing technologies was subsequently coined <a href="//en.wikipedia.org/wiki/Hybrid_genome_assembly" title="Hybrid genome assembly"><i>hybrid assembly</i></a>.</p>
<p>From 2006, the <a href="//en.wikipedia.org/wiki/Illumina_(company)" title="Illumina (company)">Illumina</a> (previously Solexa) technology has been available and can generate about 100 million reads per run on a single sequencing machine. Compare this to the 35 million reads of the human genome project which needed several years to be produced on hundreds of sequencing machines. Illumina was initially limited to a length of only 36 bases, making it less suitable for de novo assembly (such as <a href="//en.wikipedia.org/wiki/De_novo_transcriptome_assembly" title="De novo transcriptome assembly">de novo transcriptome assembly</a>), but newer iterations of the technology achieve read lengths above 100 bases from both ends of a 3-400bp clone. Announced at the end of 2007, the SHARCGS assembler<sup id="cite_ref-5" class="reference"><a href="#cite_note-5"><span>[</span>5<span>]</span></a></sup> by Dohm et al. was the first published assembler that was used for an assembly with Solexa reads. It was quickly followed by a number of others.</p>
<p>Later, new technologies like <a href="//en.wikipedia.org/wiki/ABI_Solid_Sequencing" title="ABI Solid Sequencing">SOLiD</a> from <a href="//en.wikipedia.org/wiki/Applied_Biosystems" title="Applied Biosystems">Applied Biosystems</a>, <a href="//en.wikipedia.org/wiki/Ion_Torrent" title="Ion Torrent" class="mw-redirect">Ion Torrent</a> and <a href="//en.wikipedia.org/wiki/SMRT_sequencing" title="SMRT sequencing" class="mw-redirect">SMRT</a> were released and new technologies (e.g. <a href="//en.wikipedia.org/wiki/Nanopore_sequencing" title="Nanopore sequencing">Nanopore sequencing</a>) continue to emerge.</p>
<h2><span class="mw-headline" id="Greedy_algorithm">Greedy algorithm</span></h2>
<p>Given a set of sequence fragments the object is to find the <a href="//en.wikipedia.org/wiki/Shortest_common_supersequence" title="Shortest common supersequence">shortest common supersequence</a>.</p>
<ol>
<li>calculate pairwise alignments of all fragments</li>
<li>choose two fragments with the largest overlap</li>
<li>merge chosen fragments</li>
<li>repeat step 2. and 3. until only one fragment is left</li>
</ol>
<p>The result is a suboptimal solution to the problem.<sup class="noprint Inline-Template Template-Fact" style="white-space:nowrap;">[<i><a href="//en.wikipedia.org/wiki/Wikipedia:Citation_needed" title="Wikipedia:Citation needed"><span title="This claim needs references to reliable sources. (January 2012)">citation needed</span></a></i>]</sup></p>
<h2><span class="mw-headline" id="Available_assemblers">Available assemblers</span></h2>
<p>The following table lists assemblers that have a de-novo assembly capability on at least one of the supported technologies.<sup id="cite_ref-6" class="reference"><a href="#cite_note-6"><span>[</span>6<span>]</span></a></sup></p>
<table class="wikitable sortable" border="1">
<tr>
<th>Name</th>
<th>Type</th>
<th>Technologies</th>
<th>Author</th>
<th>Presented /
<p>Last updated</p>
</th>
<th>Licence*</th>
<th>Homepage</th>
</tr>
<tr>
<td>ABySS</td>
<td>(large) genomes</td>
<td>Solexa, SOLiD</td>
<td>Simpson, J. et al.</td>
<td>2008 / 2011</td>
<td>NC-A</td>
<td><a rel="nofollow" class="external text" href="http://www.bcgsc.ca/platform/bioinfo/software/abyss">link</a></td>
</tr>
<tr>
<td>ALLPATHS-LG</td>
<td>(large) genomes</td>
<td>Solexa, SOLiD</td>
<td>Gnerre, S. et al.</td>
<td>2011</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://www.broadinstitute.org/science/programs/genome-biology/crd">link</a></td>
</tr>
<tr>
<td>AMOS</td>
<td>genomes</td>
<td>Sanger, 454</td>
<td>Salzberg, S. et al.</td>
<td>2002? / 2008?</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://sourceforge.net/apps/mediawiki/amos/index.php?title=AMOS">link</a></td>
</tr>
<tr>
<td>Arapan-M</td>
<td>Medium Genomes (e.g. E.coli)</td>
<td>All</td>
<td>Sahli, M. &amp; Shibuya, T.</td>
<td>2011 / 2012</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://sourceforge.net/projects/dnascissor/files">link</a></td>
</tr>
<tr>
<td>Arapan-S</td>
<td>Small Genomes (Viruses and Bacteria)</td>
<td>All</td>
<td>Sahli, M. &amp; Shibuya, T.</td>
<td>2011 / 2012</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://sourceforge.net/projects/dnascissor/files">link</a></td>
</tr>
<tr>
<td><a href="//en.wikipedia.org/wiki/Celera" title="Celera" class="mw-redirect">Celera</a> WGA Assembler / CABOG</td>
<td>(large) genomes</td>
<td>Sanger, 454, Solexa</td>
<td>Myers, G. et al.; Miller G. et al.</td>
<td>2004 / 2010</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://www.jcvi.org/cms/research/projects/celera-assembler/overview/">link</a></td>
</tr>
<tr>
<td>CLC Genomics Workbench &amp; CLC Assembly Cell</td>
<td>genomes</td>
<td>Sanger, 454, Solexa, SOLiD</td>
<td><a href="//en.wikipedia.org/wiki/CLC_bio" title="CLC bio">CLC bio</a></td>
<td>2008 / 2010 / 2011</td>
<td>C</td>
<td><a rel="nofollow" class="external text" href="http://www.clcbio.com/products/">link</a></td>
</tr>
<tr>
<td>Cortex</td>
<td>genomes</td>
<td>Solexa, SOLiD</td>
<td>Iqbal, Z. <i>et al.</i></td>
<td>2011</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://cortexassembler.sourceforge.net/">link</a></td>
</tr>
<tr>
<td>DNA Baser</td>
<td>genomes</td>
<td>Sanger, 454</td>
<td>Heracle BioSoft SRL</td>
<td>01.2014</td>
<td>C</td>
<td>www.DnaBaser.com</td>
</tr>
<tr>
<td>DNA Dragon</td>
<td>genomes</td>
<td>Illumina, SOLiD, Complete Genomics, 454, Sanger</td>
<td>SequentiX</td>
<td>2011</td>
<td>C</td>
<td><a rel="nofollow" class="external text" href="https://www.dna-dragon.com/">link</a></td>
</tr>
<tr>
<td>DNAnexus</td>
<td>genomes</td>
<td>Illumina, SOLiD, Complete Genomics</td>
<td>DNAnexus</td>
<td>2011</td>
<td>C</td>
<td><a rel="nofollow" class="external text" href="https://dnanexus.com/">link</a></td>
</tr>
<tr>
<td>Edena</td>
<td>genomes</td>
<td>Illumina</td>
<td>D. Hernandez, P. François, L. Farinelli, M. Osteras, and J. Schrenzel.</td>
<td>2008/2013</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://www.genomic.ch/edena.php">link</a></td>
</tr>
<tr>
<td>Euler</td>
<td>genomes</td>
<td>Sanger, 454 (,Solexa&#160;?)</td>
<td>Pevzner, P. et al.</td>
<td>2001 / 2006?</td>
<td>(C / NC-A?)</td>
<td><a rel="nofollow" class="external text" href="http://nbcr.sdsc.edu/euler/">link</a></td>
</tr>
<tr>
<td>Euler-sr</td>
<td>genomes</td>
<td>454, Solexa</td>
<td>Chaisson, MJ. et al.</td>
<td>2008</td>
<td>NC-A</td>
<td><a rel="nofollow" class="external text" href="http://euler-assembler.ucsd.edu/portal/">link</a></td>
</tr>
<tr>
<td>Forge</td>
<td>(large) genomes, EST, metagenomes</td>
<td>454, Solexa, SOLID, Sanger</td>
<td>Platt, DM, Evers, D.</td>
<td>2010</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://combiol.org/forge/">link</a></td>
</tr>
<tr>
<td><a href="//en.wikipedia.org/wiki/Geneious" title="Geneious">Geneious</a></td>
<td>genomes</td>
<td>Sanger, 454, Solexa, Ion Torrent, Complete Genomics, PacBio, Oxford Nanopore, Illumina</td>
<td><a href="//en.wikipedia.org/wiki/Biomatters" title="Biomatters">Biomatters Ltd</a></td>
<td>2009 / 2013</td>
<td>C</td>
<td><a rel="nofollow" class="external text" href="http://geneious.com/">link</a></td>
</tr>
<tr>
<td>Graph Constructor</td>
<td>(large) genomes</td>
<td>Sanger, 454, Solexa, SOLiD</td>
<td><a href="//en.wikipedia.org/w/index.php?title=Convey_Computer&amp;action=edit&amp;redlink=1" class="new" title="Convey Computer (page does not exist)">Convey Computer Corporation</a></td>
<td>2011</td>
<td>C</td>
<td><a rel="nofollow" class="external text" href="http://www.conveycomputer.com/lifesciences/">link</a></td>
</tr>
<tr>
<td>IDBA (Iterative De Bruijn graph short read Assembler)</td>
<td>(large) genomes</td>
<td>Sanger,454,Solexa</td>
<td>Yu Peng, Henry C. M. Leung, Siu-Ming Yiu, <a href="//en.wikipedia.org/wiki/Y._L._Chin" title="Y. L. Chin" class="mw-redirect">Francis Y. L. Chin</a></td>
<td>2010</td>
<td>(C / NC-A?)</td>
<td><a rel="nofollow" class="external text" href="http://www.cs.hku.hk/~alse/idba/">link</a></td>
</tr>
<tr>
<td>LIGR Assembler (derived from TIGR Assembler)</td>
<td>genomic</td>
<td>Sanger</td>
<td>-</td>
<td>2009/ 2012</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://sourceforge.net/projects/ligr-assembler/">link</a></td>
</tr>
<tr>
<td>MaSuRCA (Maryland Super Read - Celera Assembler)</td>
<td>(large) genomes</td>
<td>Sanger, Illumina, 454</td>
<td>Aleksey Zimin, Guillaume Marçais, Daniela Puiu, Michael Roberts, Steven L. Salzberg, James A. Yorke</td>
<td>2012 / 2013</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://www.genome.umd.edu/masurca.html">link</a></td>
</tr>
<tr>
<td>MIRA (Mimicking Intelligent Read Assembly)</td>
<td>genomes, ESTs</td>
<td>Sanger, 454, Solexa</td>
<td>Chevreux, B.</td>
<td>1998 / 2011</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://sourceforge.net/apps/mediawiki/mira-assembler/">link</a></td>
</tr>
<tr>
<td>NextGENe</td>
<td>(small genomes?)</td>
<td>454, Solexa, SOLiD</td>
<td>Softgenetics</td>
<td>2008</td>
<td>C</td>
<td><a rel="nofollow" class="external text" href="http://softgenetics.com/NextGENe.html">link</a></td>
</tr>
<tr>
<td><a href="//en.wikipedia.org/wiki/Newbler" title="Newbler">Newbler</a></td>
<td>genomes, ESTs</td>
<td>454, Sanger</td>
<td>454/Roche</td>
<td>2009</td>
<td>C</td>
<td><a rel="nofollow" class="external text" href="http://www.454.com/">link</a></td>
</tr>
<tr>
<td>PADENA</td>
<td>genomes</td>
<td>454, Sanger</td>
<td>454/Roche</td>
<td>2010</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://bio.codeplex.com/">link</a></td>
</tr>
<tr>
<td>PASHA</td>
<td>(large) genomes</td>
<td>Illumina</td>
<td>Liu, Schmidt, Maskell</td>
<td>2011</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://sites.google.com/site/yongchaosoftware/pasha">link</a></td>
</tr>
<tr>
<td><a href="//en.wikipedia.org/wiki/Phrap" title="Phrap">Phrap</a></td>
<td>genomes</td>
<td>Sanger, 454, Solexa</td>
<td>Green, P.</td>
<td>1994 / 2008</td>
<td>C / NC-A</td>
<td><a rel="nofollow" class="external text" href="http://www.phrap.org/">link</a></td>
</tr>
<tr>
<td><a href="//en.wikipedia.org/wiki/The_Institute_for_Genomic_Research" title="The Institute for Genomic Research" class="mw-redirect">TIGR</a> Assembler</td>
<td>genomic</td>
<td>Sanger</td>
<td>-</td>
<td>1995 / 2003</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="ftp://ftp.jcvi.org/pub/software/assembler/">link</a></td>
</tr>
<tr>
<td>Ray<sup id="cite_ref-7" class="reference"><a href="#cite_note-7"><span>[</span>7<span>]</span></a></sup></td>
<td>genomes</td>
<td>Illumina, mix of Illumina and 454, paired or not</td>
<td>Sébastien Boisvert, François Laviolette &amp; Jacques Corbeil.</td>
<td>2010</td>
<td>OS [GNU General Public License]</td>
<td><a rel="nofollow" class="external text" href="http://denovoassembler.sf.net/">link</a></td>
</tr>
<tr>
<td>Sequencher</td>
<td>genomes</td>
<td>traditional and next generation sequence data</td>
<td><a href="//en.wikipedia.org/wiki/Gene_Codes_Corporation" title="Gene Codes Corporation">Gene Codes Corporation</a></td>
<td>1991 / 2009 / 2011</td>
<td>C</td>
<td><a rel="nofollow" class="external text" href="http://www.genecodes.com/">link</a></td>
</tr>
<tr>
<td>SeqMan NGen</td>
<td>(large) genomes, exomes, transcriptomes, metagenomes, ESTs</td>
<td>Illumina, ABI SOLiD, Roche 454, Ion Torrent, Solexa, Sanger</td>
<td><a href="//en.wikipedia.org/wiki/DNASTAR" title="DNASTAR">DNASTAR</a></td>
<td>2007 / 2011</td>
<td>C</td>
<td><a rel="nofollow" class="external text" href="http://www.dnastar.com/t-products-seqman-ngen.aspx">link</a></td>
</tr>
<tr>
<td>SGA</td>
<td>(large) genomes</td>
<td>Illumina, Sanger (Roche 454?, Ion Torrent?)</td>
<td>Simpson, J.T. et al.</td>
<td>2011 / 2012</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="https://github.com/jts/sga">link</a></td>
</tr>
<tr>
<td>SHARCGS</td>
<td>(small) genomes</td>
<td>Solexa</td>
<td>Dohm et al.</td>
<td>2007 / 2007</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://sharcgs.molgen.mpg.de/">link</a></td>
</tr>
<tr>
<td>SOPRA</td>
<td>genomes</td>
<td>Illumina, SOLiD, Sanger, 454</td>
<td>Dayarian, A. et al.</td>
<td>2010 / 2011</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://www.physics.rutgers.edu/~anirvans/SOPRA/">link</a></td>
</tr>
<tr>
<td>SparseAssembler</td>
<td>(large) genomes</td>
<td>Illumina, 454, Ion torrent</td>
<td>Ye, C. et al.</td>
<td>2012 / 2012</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="https://sites.google.com/site/sparseassembler/">link</a></td>
</tr>
<tr>
<td>SSAKE</td>
<td>(small) genomes</td>
<td>Solexa (SOLiD? Helicos?)</td>
<td>Warren, R. et al.</td>
<td>2007 / 2007</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://www.bcgsc.ca/platform/bioinfo/software/ssake">link</a></td>
</tr>
<tr>
<td>SOAPdenovo</td>
<td>genomes</td>
<td>Solexa</td>
<td>Li, R. et al.</td>
<td>2009 / 2009</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://soap.genomics.org.cn/soapdenovo.html">link</a></td>
</tr>
<tr>
<td><a href="//en.wikipedia.org/wiki/SPAdes_(software)" title="SPAdes (software)">SPAdes</a></td>
<td>(small) genomes, single-cell</td>
<td>Illumina, Solexa</td>
<td>Bankevich, A et al.</td>
<td>2012 / 2013</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://bioinf.spbau.ru/en/spades">link</a></td>
</tr>
<tr>
<td>Staden gap4 package</td>
<td>BACs (, small genomes?)</td>
<td>Sanger</td>
<td>Staden et al.</td>
<td>1991 / 2008</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://staden.sourceforge.net/">link</a></td>
</tr>
<tr>
<td>Taipan</td>
<td>(small) genomes</td>
<td>Illumina</td>
<td>Schmidt, B. <i>et al.</i></td>
<td>2009</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://sourceforge.net/projects/taipan/">link</a></td>
</tr>
<tr>
<td>VCAKE</td>
<td>(small) genomes</td>
<td>Solexa (SOLiD?, Helicos?)</td>
<td>Jeck, W. et al.</td>
<td>2007 / 2007</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://sourceforge.net/projects/vcake">link</a></td>
</tr>
<tr>
<td>Phusion assembler</td>
<td>(large) genomes</td>
<td>Sanger</td>
<td>Mullikin JC, <i>et al.</i></td>
<td>2003</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://www.sanger.ac.uk/Software/production/phusion/">link</a></td>
</tr>
<tr>
<td>Quality Value Guided SRA (QSRA)</td>
<td>genomes</td>
<td>Sanger, Solexa</td>
<td>Bryant DW, <i>et al.</i></td>
<td>2009</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://qsra.cgrb.oregonstate.edu/">link</a></td>
</tr>
<tr>
<td><a href="//en.wikipedia.org/wiki/Velvet_assembler" title="Velvet assembler">Velvet</a></td>
<td>(small) genomes</td>
<td>Sanger, 454, Solexa, SOLiD</td>
<td>Zerbino, D. et al.</td>
<td>2007 / 2009</td>
<td>OS</td>
<td><a rel="nofollow" class="external text" href="http://www.ebi.ac.uk/~zerbino/velvet/">link</a></td>
</tr>
<tr>
<td colspan="7" style="border-top: 1px solid #333;"><small>*<b>Licences:</b> OS = Open Source; C = Commercial; C / NC-A = Commercial but free for non-commercial and academics; Brackets = unclear, but most likely C / NC-A</small></td>
</tr>
</table>
<h2><span class="mw-headline" id="See_also">See also</span></h2>
<ul>
<li><a href="//en.wikipedia.org/wiki/Sequence_alignment" title="Sequence alignment">Sequence alignment</a></li>
<li><a href="//en.wikipedia.org/wiki/Genome_assembly" title="Genome assembly" class="mw-redirect">Genome assembly</a></li>
<li><a href="//en.wikipedia.org/wiki/De_novo_transcriptome_assembly" title="De novo transcriptome assembly">De novo transcriptome assembly</a></li>
<li><a href="//en.wikipedia.org/wiki/Set_cover_problem" title="Set cover problem">Set cover problem</a></li>
</ul>
<h2><span class="mw-headline" id="References">References</span></h2>
<div class="reflist columns references-column-width" style="-moz-column-width: 30em; -webkit-column-width: 30em; column-width: 30em; list-style-type: decimal;">
<ol class="references">
<li id="cite_note-1"><span class="mw-cite-backlink"><b><a href="#cite_ref-1">^</a></b></span> <span class="reference-text"><span class="citation journal">Myers, E. W.; Sutton, GG; Delcher, AL; Dew, IM; Fasulo, DP; Flanigan, MJ; Kravitz, SA; Mobarry, CM et al. (March 2000). <a rel="nofollow" class="external text" href="http://www.sciencemag.org/cgi/pmidlookup?view=long&amp;pmid=10731133">"A whole-genome assembly of Drosophila"</a>. <i>Science</i> <b>287</b> (5461): 2196–204. <a href="//en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1126%2Fscience.287.5461.2196">10.1126/science.287.5461.2196</a>. <a href="//en.wikipedia.org/wiki/PubMed_Identifier" title="PubMed Identifier" class="mw-redirect">PMID</a>&#160;<a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pubmed/10731133">10731133</a>.</span><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ASequence+assembly&amp;rft.atitle=A+whole-genome+assembly+of+Drosophila&amp;rft.au=Adams%2C+M.+D.&amp;rft.au=Anson%2C+E.+L.&amp;rft.au=Beasley%2C+E.+M.&amp;rft.au=Bolanos%2C+R.+A.&amp;rft.au=Brandon%2C+R.+C.&amp;rft.au=Chen%2C+L&amp;rft.au=Chou%2C+H.+H.&amp;rft.au=Delcher%2C+AL&amp;rft.au=Dew%2C+IM&amp;rft.au=Dunn%2C+P.+J.&amp;rft.au=Fasulo%2C+DP&amp;rft.aufirst=E.+W.&amp;rft.au=Flanigan%2C+MJ&amp;rft.au=Halpern%2C+A.+L.&amp;rft.au=Jordan%2C+C.+M.&amp;rft.au=Kravitz%2C+SA&amp;rft.au=Lai%2C+Z&amp;rft.aulast=Myers&amp;rft.au=Liang%2C+Y&amp;rft.au=Lonardi%2C+S&amp;rft.au=Mobarry%2C+CM&amp;rft.au=Myers%2C+E.+W.&amp;rft.au=Nusskern%2C+D.+R.&amp;rft.au=Reinert%2C+KH&amp;rft.au=Remington%2C+K.+A.&amp;rft.au=Rubin%2C+G.+M.&amp;rft.au=Sutton%2C+GG&amp;rft.au=Venter%2C+J.+C.&amp;rft.au=Zhan%2C+M&amp;rft.au=Zhang%2C+Q&amp;rft.au=Zheng%2C+X&amp;rft.date=March+2000&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Fwww.sciencemag.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D10731133&amp;rft_id=info%3Adoi%2F10.1126%2Fscience.287.5461.2196&amp;rft_id=info%3Apmid%2F10731133&amp;rft.issue=5461&amp;rft.jtitle=Science&amp;rft.pages=2196-204&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=287" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-2"><span class="mw-cite-backlink"><b><a href="#cite_ref-2">^</a></b></span> <span class="reference-text"><span class="citation journal">Batzoglou, S.; Jaffe, DB; Stanley, K; Butler, J; Gnerre, S; Mauceli, E; Berger, B; Mesirov, JP; Lander, ES (January 2002). <a rel="nofollow" class="external text" href="http://www.genome.org/cgi/pmidlookup?view=long&amp;pmid=11779843">"ARACHNE: a whole-genome shotgun assembler"</a>. <i>Genome Research</i> <b>12</b> (1): 177–89. <a href="//en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1101%2Fgr.208902">10.1101/gr.208902</a>. <a href="//en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC155255">155255</a>. <a href="//en.wikipedia.org/wiki/PubMed_Identifier" title="PubMed Identifier" class="mw-redirect">PMID</a>&#160;<a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pubmed/11779843">11779843</a>.</span><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ASequence+assembly&amp;rft.atitle=ARACHNE%3A+a+whole-genome+shotgun+assembler&amp;rft.au=Batzoglou%2C+S.&amp;rft.au=Berger%2C+B&amp;rft.au=Butler%2C+J&amp;rft.aufirst=S.&amp;rft.au=Gnerre%2C+S&amp;rft.au=Jaffe%2C+DB&amp;rft.au=Lander%2C+ES&amp;rft.aulast=Batzoglou&amp;rft.au=Mauceli%2C+E&amp;rft.au=Mesirov%2C+JP&amp;rft.au=Stanley%2C+K&amp;rft.date=January+2002&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Fwww.genome.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D11779843&amp;rft_id=info%3Adoi%2F10.1101%2Fgr.208902&amp;rft_id=info%3Apmc%2F155255&amp;rft_id=info%3Apmid%2F11779843&amp;rft.issue=1&amp;rft.jtitle=Genome+Research&amp;rft.pages=177-89&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=12" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-3"><span class="mw-cite-backlink"><b><a href="#cite_ref-3">^</a></b></span> <span class="reference-text"><a rel="nofollow" class="external text" href="http://amos.sourceforge.net/">AMOS page</a> with links to various papers</span></li>
<li id="cite_note-4"><span class="mw-cite-backlink"><b><a href="#cite_ref-4">^</a></b></span> <span class="reference-text">Copy in Google groups of the <a rel="nofollow" class="external text" href="http://groups.google.com/group/bionet.software/browse_thread/thread/b34b348011d04f0e?fwc=1">post announcing MIRA 2.9.8 hybrid version</a> in the bionet.software Usenet group</span></li>
<li id="cite_note-5"><span class="mw-cite-backlink"><b><a href="#cite_ref-5">^</a></b></span> <span class="reference-text"><span class="citation journal">Dohm, J. C.; Lottaz, C.; Borodina, T.; Himmelbauer, H. (November 2007). <a rel="nofollow" class="external text" href="http://www.genome.org/cgi/pmidlookup?view=long&amp;pmid=17908823">"SHARCGS, a fast and highly accurate short-read assembly algorithm for de novo genomic sequencing"</a>. <i>Genome Research</i> <b>17</b> (11): 1697–706. <a href="//en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1101%2Fgr.6435207">10.1101/gr.6435207</a>. <a href="//en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC2045152">2045152</a>. <a href="//en.wikipedia.org/wiki/PubMed_Identifier" title="PubMed Identifier" class="mw-redirect">PMID</a>&#160;<a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pubmed/17908823">17908823</a>.</span><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ASequence+assembly&amp;rft.atitle=SHARCGS%2C+a+fast+and+highly+accurate+short-read+assembly+algorithm+for+de+novo+genomic+sequencing&amp;rft.au=Borodina%2C+T.&amp;rft.au=Dohm%2C+J.+C.&amp;rft.aufirst=J.+C.&amp;rft.au=Himmelbauer%2C+H.&amp;rft.aulast=Dohm&amp;rft.au=Lottaz%2C+C.&amp;rft.date=November+2007&amp;rft.genre=article&amp;rft_id=http%3A%2F%2Fwww.genome.org%2Fcgi%2Fpmidlookup%3Fview%3Dlong%26pmid%3D17908823&amp;rft_id=info%3Adoi%2F10.1101%2Fgr.6435207&amp;rft_id=info%3Apmc%2F2045152&amp;rft_id=info%3Apmid%2F17908823&amp;rft.issue=11&amp;rft.jtitle=Genome+Research&amp;rft.pages=1697-706&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=17" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
<li id="cite_note-6"><span class="mw-cite-backlink"><b><a href="#cite_ref-6">^</a></b></span> <span class="reference-text"><a rel="nofollow" class="external text" href="http://seqanswers.com/forums/showthread.php?t=43">list of software including mapping assemblers in the SeqAnswers discussion forum.</a></span></li>
<li id="cite_note-7"><span class="mw-cite-backlink"><b><a href="#cite_ref-7">^</a></b></span> <span class="reference-text"><span class="citation journal">Boisvert, Sébastien; Laviolette, François; Corbeil, Jacques (October 2010). <a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3119603">"Ray: simultaneous assembly of reads from a mix of high-throughput sequencing technologies"</a>. <i>Journal of Computational Biology</i> <b>17</b> (11): 1519–33. <a href="//en.wikipedia.org/wiki/Digital_object_identifier" title="Digital object identifier">doi</a>:<a rel="nofollow" class="external text" href="http://dx.doi.org/10.1089%2Fcmb.2009.0238">10.1089/cmb.2009.0238</a>. <a href="//en.wikipedia.org/wiki/PubMed_Central" title="PubMed Central">PMC</a>&#160;<a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pmc/articles/PMC3119603">3119603</a>. <a href="//en.wikipedia.org/wiki/PubMed_Identifier" title="PubMed Identifier" class="mw-redirect">PMID</a>&#160;<a rel="nofollow" class="external text" href="//www.ncbi.nlm.nih.gov/pubmed/20958248">20958248</a>.</span><span title="ctx_ver=Z39.88-2004&amp;rfr_id=info%3Asid%2Fen.wikipedia.org%3ASequence+assembly&amp;rft.atitle=Ray%3A+simultaneous+assembly+of+reads+from+a+mix+of+high-throughput+sequencing+technologies&amp;rft.au=Boisvert%2C+S%C3%A9bastien&amp;rft.au=Corbeil%2C+Jacques&amp;rft.aufirst=S%C3%A9bastien&amp;rft.aulast=Boisvert&amp;rft.au=Laviolette%2C+Fran%C3%A7ois&amp;rft.date=October+2010&amp;rft.genre=article&amp;rft_id=%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC3119603&amp;rft_id=info%3Adoi%2F10.1089%2Fcmb.2009.0238&amp;rft_id=info%3Apmc%2F3119603&amp;rft_id=info%3Apmid%2F20958248&amp;rft.issue=11&amp;rft.jtitle=Journal+of+Computational+Biology&amp;rft.pages=1519-33&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.volume=17" class="Z3988"><span style="display:none;">&#160;</span></span></span></li>
</ol>
</div>


<!-- 
NewPP limit report
Parsed by mw1178
CPU time usage: 0.308 seconds
Real time usage: 0.345 seconds
Preprocessor visited node count: 658/1000000
Preprocessor generated node count: 3463/1500000
Post‐expand include size: 14017/2048000 bytes
Template argument size: 560/2048000 bytes
Highest expansion depth: 11/40
Expensive parser function count: 1/500
Lua time usage: 0.041/10.000 seconds
Lua memory usage: 1.55 MB/50 MB
-->
